packages/eslint-plugin/src/rules/no-use-before-define.ts

```diff
@@ -219,7 +242,7 @@ export default util.createRule<Options, MessageIds>({
       variable: TSESLint.Scope.Variable,
       reference: TSESLint.Scope.Reference,
     ): boolean {
-      if (reference.isTypeReference && options.ignoreTypeReferences) {
+      if (options.ignoreTypeReferences && isTypeReference(reference)) {
```

Comment:
I've reversed the conditions, if `options.ignoreTypeReferences` is false we'll just shortcircuit out of this and prevent unnecessary computation.